tests/averaging/rotation/test_shonan.py

```diff
@@ -119,6 +119,37 @@ def test_pickleable(self):
         except TypeError:
             self.fail("Cannot dump rotation averaging object using pickle")
 
+    def test_nonconsecutive_indices(self):
```

Comment:
might also be useful to add a test with edges that dont belong to a connected component.

Reply:
So rotation averaging runs after pruning to the largest component in `MultiviewOptimizer`, so that case should already be handled actually (in other words, we won't allow such input).

Reply:
okay